src/lib/transformers.ts

```diff
@@ -106,55 +106,46 @@ export function batcher<T>(
  * this special case. To recover from the error, have `handleError()` return normally.
  * To fail, simply re-throw the provided error (or a new, more appropriate Error).
  *
- * Example:
+ * @example
  * // Without endCatcher()
- * const source = new Stream<string>();
- * source.write("Craig");
- * source.write("Jolene");
- * source.write("Sam");
- * source.write("Cassandra");
- * source.write("Xavier");
- * source.end().catch((e) => {
- *   if (e.message === "Too many guests!") {
- *     requestBiggerTable();
- *   } else {
- *     throw e;
- *   }
- * });
- *
- * source.forEach(
- *   addGuestToReservation,
- *   () => {
- *     if (guestsAdded() > 4) {
- *       throw new Error("Too many guests!")
+ * getRowsFromDatabase<MyObject>(myQuery)
+ *   .transform(
+ *     // other transforms etc
+ *   )
+ *   .forEach(
+ *     process,
+ *     () => {
+ *       if (someCondition) {
+ *         throw new NotAllRowsProcessedError();
+ *       }
  *     }
- *   }
- * ); // This throws an unhandled promise rejection error, even though the source handled it!
+ *   ); // This throws an unhandled promise rejection error, even if the source handles it!
```

Comment:
This sounds more scary than it really is. Perhaps:
`// This throws a rejection, when the source uses the default behavior of bouncing an error into result`